src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value, null, tail);
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        validatePositionIndex(index);
+        if (head == null) {
+            Node<T> node = new Node<>(value, null, null);
+            head = tail = node;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(value, head, null);
+            head = newNode;
+        } else if (index == size) {
+            add(value);
+            return;
+        } else {
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        } 
```
put this case separately, we don't need else after return